File: aeds02/units/u03/Notacoes/ex/PalindromoIterativo.java
public class PalindromoIterativo {
	public static boolean isFim(String str) {
		return str.length() == 3 && str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M'; 
	}

	public static String getPalindromo(String str) {
		return isPalindromo(str) ? "SIM" : "NAO";
	}

	public static boolean isPalindromo(String str) {
		boolean resp = true;
		int j = str.length() - 1;

		for(int i = 0; i < str.length() / 2; i++) {
			if(str.charAt(i) != str.charAt(j)) {
				resp = false;
				i = str.length();
			}
			j--;
		}

		return resp;
	}

	public static void main(String[] args) {
		String[] entrada = new String[1000];
		String[] saida = new String[1000];
		int numEntrada = 0;

		do {
			entrada[numEntrada] = MyIO.readLine();
			saida[numEntrada] = getPalindromo(entrada[numEntrada]);
		} while(!isFim(entrada[numEntrada++]));
		numEntrada--;

		for(int i = 0; i < numEntrada; i++) {
			MyIO.println(saida[i]);
		}
	}
}

File: aeds02/tps/tp1/PalindromoIterativo.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>

#define MAXTAM 1000
#define true 1
#define false 0

int isFim(char *str);
int isPalindromo(char *str);
char* getResp(char *str);

int isFim(char *str) {
	return strlen(str) == 4 && str[0] == 'F' && str[1] == 'I' && str[2] == 'M' && str[3] == '\n';
}

char* getResp(char *str) {
	return isPalindromo(str) == true ? "SIM" : "NAO";
}

int isPalindromo(char *str) {
	int resp = true;
	int j = (int)strlen(str) - 2;

	for(int i = 0; i < (int)strlen(str) / 2; i++) {
		if(str[i] != str[j]) {
			resp = false;
			i = (int)strlen(str);
		}
		j--;
	}

	return resp;
}

int main() {
	char **entrada = (char**) malloc(MAXTAM * sizeof(char*));
	char **saida = (char**) malloc(MAXTAM * sizeof(char*));
	int numEntrada = 0;

	do {
		entrada[numEntrada] = (char*) malloc(MAXTAM * sizeof(char*));
		saida[numEntrada] = (char*) malloc(MAXTAM * sizeof(char*));

		entrada[numEntrada] = fgets(entrada[numEntrada], MAXTAM, stdin);
		saida[numEntrada] = getResp(entrada[numEntrada]);
		
	} while(isFim(entrada[numEntrada++]) != true);
	numEntrada--;
	
	for(int i = 0; i < numEntrada; i++) {
		printf("%s\n", saida[i]);
	}

	//free(entrada);
	//free(saida);

	return 0;
}
